fcrepo-jms/src/test/java/org/fcrepo/integration/jms/observer/HeadersJMSIT.java

```diff
@@ -131,7 +134,8 @@ public void testIngestion() throws RepositoryException {
         try {
             containerService.findOrCreate(session, testIngested);
             session.save();
-            awaitMessageOrFail(testIngested, NODE_ADDED_EVENT_TYPE, null);
+            final Message ingestMsg = awaitMessageOrFail(testIngested, NODE_ADDED_EVENT_TYPE);
+            assertTrue(getResourceTypes(ingestMsg).contains("fedora:Resource"));
```

Comment:
They could, but that would be wrong for the other reason I gave, which is why I'm not going to do that. I'd much rather put this on hold and deal properly with the namespace question, rather than gunk up the internals of the reference kernel implementation for short-term gain.
As for the two needs in specific, as I said, I don't recognize the first at all, but I'm willing to treat with it at your insistence, and the second seems far less important to me than I think it may seem to you. Neither of them is much of a priority to me, which is why I would rather wait.